### PR TITLE
[KED-824] Adopt a new JSON format

### PR DESCRIPTION
**Pull request by tolomea**
## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking “Submit Pull Request”:

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

## Motivation and Context
Why was this PR created?

## How has this been tested?
What testing strategies have you used?

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes
- [x] Assigned myself to the PR
- [x] Added `Type` label to the PR

### CODE REVIEW

File: package/kedro_viz/server.py [hunk 1]
```diff
@@ -21,7 +22,7 @@ def root():
 
 
 @app.route("/logs/nodes.json")
-def nodes():
+def nodes_old():
```
Comment (idanov) [inline package/kedro_viz/server.py]: Could we change the name to something like:
```suggestion
def nodes_deprecated():
```

File: package/kedro_viz/server.py [hunk 2]
```diff
@@ -37,6 +38,62 @@ def nodes():
...
+                "type": "task",
+                "id": task_id,
+                "name": node.short_name,
+                "full_name": str(node),
+                "tags": sorted(node.tags),
+            }
+        )
+        all_tags.update(node.tags)
+        for data_set in node.inputs:
+            namespace = data_set.split("@")[0]
```
Comment (idanov) [inline package/kedro_viz/server.py]: Is the term  `namespace` here the correct one? WDYT @LorenaBalanQB ?

File: package/tests/test_server.py [hunk 3]
```diff
@@ -10,12 +10,9 @@
 
 from kedro_viz import server
 
-EXPECTED_PIPELINE_DATA = [
+EXPECTED_PIPELINE_DATA_OLD = [
```
Comment (idanov) [inline package/tests/test_server.py]: ⬆️ Same as the comment for `nodes_old`.

File: package/tests/test_server.py [hunk 4]
```diff
@@ -49,6 +43,150 @@
 ]
 
 
+EXPECTED_PIPELINE_DATA = {
```
Comment (idanov) [inline package/tests/test_server.py]: Is it possible to come up with a more minimalistic test case in order to be easier to change later?

File: package/tests/test_server.py [hunk 5]
```diff
@@ -146,9 +293,17 @@ def test_root_endpoint(client):
     assert "Kedro Viz" in response.data.decode()
 
 
-def test_nodes_endpoint(client):
+def test_old_nodes_endpoint(client):
```
Comment (idanov) [inline package/tests/test_server.py]: ⬆️

File: package/kedro_viz/server.py [hunk 6]
```diff
@@ -32,11 +33,67 @@ def nodes():
...
+@app.route("/api/nodes.json")
+def nodes_json():
+    """Serve the pipeline data."""
+
+    def pretty_name(name):
+        name = name.replace("-", " ").replace("_", " ")
+        parts = [n[0].upper() + n[1:] for n in name.split()]
+        return " ".join(parts)
+
+    pipeline = get_project_context("create_pipeline")()
```
Comment (921kiyo) [inline package/kedro_viz/server.py]: What would happen if the passed pipeline is broken? Would it fail somewhere inside this function?

File: package/kedro_viz/server.py [hunk 7]
```diff
@@ -21,7 +22,7 @@ def root():
 
 
 @app.route("/logs/nodes.json")
-def nodes():
+def nodes_deprecated():
```
[the remark this answers is not in the document]
Comment (tolomea) [reply package/kedro_viz/server.py]: users don't use this, the JS does, retaining the old one is purely so the Python and JS doesn't have to merge at the same time

File: package/kedro_viz/server.py (hunk 6 above)
Comment (tolomea) [reply package/kedro_viz/server.py]: it's up to the Pipeline constructor to ensure the constructed pipeline is valid

**Pull request merged**